pkg/controllers/kluctldeployment_controller.go

```diff
@@ -318,25 +319,26 @@ func (r *KluctlDeploymentReconciler) buildFinalStatus(ctx context.Context, obj *
 	}
 
 	if lastDeployResult == nil {
-		return "deployment status unknown", kluctlv1.DeployFailedReason
+		return "deployment status unknown", kluctlv1.DeployFailedReason, time.Now().Unix()
```

Comment:
Returning time.Now() here will result in the returned value to be different on each invocation. I'd suggest to change the return type to `*time.Time` and return `nil` instead. And then, only set the new metric/gauge when the time is non-nil. This way, you'd preserve old values even if some error occurs.